apps/prairielearn/src/tests/workspaceDynamicFiles.test.ts

```diff
@@ -2,175 +2,202 @@ import * as fs from 'node:fs/promises';
-    config.workspaceEnable = false;
-    config.workspaceHomeDirRoot = await fs.mkdtemp('/tmp/prairielearn-workspace-test-');
-  });
-  after('restore configuration values', async () => {
-    await fs.rm(config.workspaceHomeDirRoot, { force: true, recursive: true });
-    config.workspaceEnable = originalConfig.workspaceEnable;
-    config.workspaceHomeDirRoot = originalConfig.workspaceHomeDirRoot;
-  });
-  before('set up testing server', helperServer.before());
-  after('shut down testing server', helperServer.after);
-
-  describe('Question with valid dynamic files', () => {
-    it('create question variant and initialize workspace', async () =>
-      createAndInitializeWorkspaceVariant('workspace'));
-
-    it('creates all static files', async () => {
-      await checkFileContents(
-        'starter_code.c',
-        '// starter_code.c\n\n#include <stdio.h>\n#include "starter_code.h"\n\nint main()\n{\n    // enter your code\n}\n',
-      );
-      await checkFileContents('starter_code.h', '// starter_code.h\n');
-      await checkFileContents('subdir/other_file.txt', 'Content in subdirectory\n');
-      await checkFileContents(
-        'binary.bin',
-        '\x00\x01\x02\x03\x04\x05\x06\x07\x08\x09\x0A\x0B\x0C\x0D\x0E\x0F',
-      );
-    });
-
-    it('creates all template files', async () => {
-      await checkFileContents('first_file.txt', 'A is WORD, while B is 35\n');
-      await checkFileContents('second_file.txt', 'The value of params.a is WORD.\n');
-      await checkFileContents('path/with/multiple/components.py', 'b = "WORD"\nprint(b)\n');
-    });
-
-    it('creates all dynamic files', async () => {
-      await checkFileContents('first_dynamic_file.py', 'a, b = b, a\nprint(a, b)\n');
-      await checkFileContents(
-        'second_dynamic_file.bin',
-        '\x00\x01\x02\x03\x04\x05\x06\x07\x08\x09\x0A\x0B\x0C\x0D\x0E\x0F\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1A\x1B\x1C\x1D\x1E\x1F',
-      );
-      await checkFileContents(
-        'path/with/multiple/dynamic/components.bin',
-        '\x00\x11\x22\x33\x44\x55\x66\x77\x88\x99\xAA\xBB\xCC\xDD\xEE\xFF',
-      );
-      await checkFileContents('blank_file.txt', '');
-      await checkFileContents('reference_file.txt', 'This is included in the workspace.\n');
-      await checkFileContents('reference_to_subdir.txt', 'Test file.\n');
-      await checkFileContents(
-        'not_normalized.txt',
-        'File identified by path that is not normalized\n',
-      );
-    });
-
-    it('template files override static files', async () => {
-      await checkFileContents('template_and_static.txt', 'The correct answer is 35.\n');
-    });
-
-    it('dynamic files override template files', async () => {
-      await checkFileContents('template_and_dynamic.csv', 'a,b\n1,1\n2,4\n3,9');
+describe('Workspace dynamic files', function () {
+  it('succeeds with valid dynamic files', async () => {
+    const targetPath = await tmp.dir({ unsafeCleanup: true });
```

Comment:
Nit: It doesn't seem like this is deleting the file on process exit. I tested locally and the temporary file still persists after the test completes.

Reply:
Resolved by 6ba59c5.